Automatic page view tracking

IeuanWalker opened the pull request:
I added the ability to track page views automatically.

I've set the view name to the classes `FullName` rather than just `Name`. This is to avoid any issues with the stats.

f.e. in my use case we have 100s of pages and a lot of them are named the same, so if we used just `name` the page view count would be a lot higher, and no way of knowing what specific page is used.

It would also mess with application insights "Funnels" and "User flow" tools if the view name is not unique.

dhindrik commented on TinyInsights/ApplicationInsightsProvider.cs:
```diff
@@ -46,6 +47,18 @@ public ApplicationInsightsProvider(string connectionString)
 
         Task.Run(SendCrashes);
+
+        if(IsAutoTrackPageViewsEnabled)
```
This will always be true here because the property is set after the constructor has finished.

dhindrik commented:
I like the idea of auto-tracking. But this way it will only track the page when it is created and not when navigating to it. For example, if you navigating back to a page it will not be tracked as a page view.

IeuanWalker commented:
@dhindrik fixed the on navigating back by subscribing to the OnAppearing event. 

I just need to do some testing to make sure it wont affect users subscribing to the same event

IeuanWalker commented:
@dhindrik I Tested and all looks to be working to me.

Added some test pages to make sure the user can still set the events, which works fine.

Also added a button on the main page to run garbage collection and output the heap size to the output, it always returned back to the base level. So no issue with memory leaks either.

IeuanWalker commented:
could potentially add the duration the user spent on the page, by subscribing to both `Appearing` and `Disappearing`. Stating a stopwatch in the `Appearing` event, then stopping the stopwatch in the `Disappearing` event, then recording the page view stat. 

But I think the PR is good as is for now, and we could look at updating that in the future.

IeuanWalker commented:
@dhindrik Sorry I messed up the rebase on this branch somehow, I created a new branch/ PR here - #18 

I've added your `WeakEventHandler` all seems to work except for the MainPage. 

If you navigate from the MainPage to another and back, the `OnAppearing` event is correctly triggered. But if you repeat it, the `OnAppearing` doesn't get triggered. From my testing the other pages all work as expected. It's like the GC is cleaning up the MainPage event even though its still being used. 

The original code worked as expected
